package.json

```diff
@@ -2,18 +2,43 @@
   "name": "@eightfold.ai/octuple",
   "version": "2.51.1",
   "license": "MIT",
-  "main": "lib/octuple.js",
+  "sideEffects": [
+    "**/*.css"
+  ],
+  "exports": {
```

Comment:
Lets add Individual components to this. That way we're set to scale for experimental and vNext components.

Reply:
Prob need to add like a public.ts in the same dir as index.ts each and only map to those exported from octuple.ts

Reply:
Don't think that will be necessary.

Do all the components we want to export individually follow a pattern like `src/components/**/index.{ts,tsx}`?
If so it's not super difficult to get these added

Reply:
After trying it out a bit, I think it's probably not necessary right now.

Additional exports can be added fairly trivially by adding:
```
preserveModules: true,
preserveModulesRoot: 'src',
```
to the output configurations in `rollup.config.mjs`, but unfortunately it doesn't seem to generate `index.mjs` and `index.cjs` files for each component, only the named files (e.g. `lib/components/Accordion/Accordion.cjs` exists, but `lib/components/Accordion/index.cjs` does not)

If that can be fixed without needing to go all out and add individual entry points for each export, then each entry point can be added like:
```json
{
    "exports": {
        "./Accordion": {
            "types": "./lib/components/Accordion/index.d.ts",
            "import": "./lib/components/Accordion/index.mjs",
            "require": "./lib/components/Accordion/index.cjs"
        }
    }
}
```

_This lengthy JSON could also be generated to avoid needing to create/update these by hand_

Reply:
Also, from looking around at other libraries (e.g. [redux](https://github.com/reduxjs/redux/blob/master/package.json), [radix-ui/primitives](https://github.com/radix-ui/primitives/blob/main/packages/react/slot/package.json)), it seems like most take the approach of either:
1. A monorepo with a `packages` directory, each exporting its own thing (components, react/svelte/<js lib> bindings, etc.)
2. A single package, but only exports the top-level `'.'` or limited additional entry points.

Reply:
@gclark-eightfold Exporting individual components as packages to NPM seems the way to go like [radix-ui/primitives](https://github.com/radix-ui/primitives/blob/main/packages/react/slot/package.json). However, the build script should be shared and updates will need to be made to the Actions to enable this. Does it need to be this complex? What we do with locale works with our current webpack config. Seems to me to be a limitation of rollup's module transpiler not considering Jest causing the 1 isssue we have found when running upstream UTs.